File: python/hash-table/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for index, val in enumerate(nums):
            if target - val in d:
                return [d[target-val], index]
            d[val] = index

        return []

File: go/array/two-sum.go
func twoSum(nums []int, target int) []int {
    var ret []int
    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums); j++ {
            if ((nums[i] + nums[j]) == target) {
                ret = append(ret,i,j)
                return ret
            }
        }
    }

    return ret
}
